The package is still marked broken. But even bypassing that it still doesn't link on aarch64-darwin for me.

```
libtool: link: ar cr .libs/libssl_iostream.a .libs/iostream-ssl.o .libs/iostream-ssl-context-cache.o .libs/iostream-ssl-test.o 
libtool: link: ranlib .libs/libssl_iostream.a
libtool: link: ( cd ".libs" && rm -f "libssl_iostream.la" && ln -s "../libssl_iostream.la" "libssl_iostream.la" )
libtool: link: ar cr .libs/libssl_iostream_openssl.a  dovecot-openssl-common.o iostream-openssl.o iostream-openssl-common.o iostream-openssl-context.o istream-openssl.o ostream-openssl.o
libtool: link: ranlib .libs/libssl_iostream_openssl.a
libtool: link: ( cd ".libs" && rm -f "libssl_iostream_openssl.la" && ln -s "../libssl_iostream_openssl.la" "libssl_iostream_openssl.la" )
/nix/store/f700nj7wlwg441h39gkq29qbviy99sgq-bash-5.3p9/bin/bash ../../libtool  --tag=CC   --mode=link clang -std=gnu23  -std=gnu99 -g -O2 -fstack-protector-strong -U_FORTIFY_SOURCE -D_FORT>

libtool: warning: Linking the executable test-iostream-ssl against the loadable module
libtool: warning: libssl_iostream_openssl.so is not portable!
libtool: link: clang -std=gnu23 -std=gnu99 -g -O2 -fstack-protector-strong -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Wall -W -Wmissing-prototypes -Wmissing-declarations -Wpointer-arith -Wchar>
ld: can't link with bundle (MH_BUNDLE) only dylibs (MH_DYLIB) file './.libs/libssl_iostream_openssl.so' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [Makefile:653: test-iostream-ssl] Error 1
make[3]: Leaving directory '/nix/var/nix/builds/nix-82060-771436163/source/src/lib-ssl-iostream'
make[2]: *** [Makefile:581: all-recursive] Error 1
make[2]: Leaving directory '/nix/var/nix/builds/nix-82060-771436163/source/src'
make[1]: *** [Makefile:712: all-recursive] Error 1
make[1]: Leaving directory '/nix/var/nix/builds/nix-82060-771436163/source'
make: *** [Makefile:556: all] Error 2
```

--- a/pkgs/by-name/do/dovecot/generic.nix
+++ b/pkgs/by-name/do/dovecot/generic.nix
@@ -4,6 +4,7 @@
   patches,
 }:
 {
+  autoreconfHook,
   stdenv,
   lib,
   fetchzip,
@@ -64,7 +65,8 @@ stdenv.mkDerivation (finalAttrs: {
     perl
     pkg-config
   ]
-  ++ lib.optionals (stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isDarwin) [ rpcsvc-proto ];
+  ++ lib.optionals (stdenv.hostPlatform.isLinux && !stdenv.hostPlatform.isDarwin) [ rpcsvc-proto ]
+  ++ lib.optionals (stdenv.hostPlatform.isDarwin) [ autoreconfHook ];
 
   buildInputs = [
     openssl
@@ -137,6 +139,12 @@ stdenv.mkDerivation (finalAttrs: {
   )
   + lib.optionalString stdenv.hostPlatform.isLinux ''
     export systemdsystemunitdir=$out/etc/systemd/system
+  ''
+  + lib.optionalString stdenv.hostPlatform.isDarwin ''
+    substituteInPlace configure.ac \
+    --replace-fail \
+      'NOPLUGIN_LDFLAGS="-no-undefined"' \
+      'NOPLUGIN_LDFLAGS="-undefined dynamic_lookup"'
   '';
 
   preBuild = lib.optionalString (lib.strings.versionOlder version "2.4" && stdenv.isDarwin) ''
@@ -203,7 +211,6 @@ stdenv.mkDerivation (finalAttrs: {
   strictDeps = true;
 
   meta = {
-    broken = lib.versionAtLeast version "2.4" && stdenv.hostPlatform.isDarwin; # fails to link openssl
     homepage = "https://dovecot.org/";
     description = "Open source IMAP and POP3 email server written with security primarily in mind";
     license = with lib.licenses; [

